api/orchestrators.go

```diff
@@ -1,29 +1,73 @@
 package api
 
 import (
+	"context"
 	"strconv"
 	"time"
 
 	"github.com/YaleSpinup/docdb-api/common"
 	"github.com/YaleSpinup/docdb-api/docdb"
 	"github.com/YaleSpinup/docdb-api/resourcegroupstaggingapi"
 	"github.com/YaleSpinup/flywheel"
+	log "github.com/sirupsen/logrus"
 )
 
 type docDBOrchestrator struct {
-	client   docdb.DocDB
-	rgClient *resourcegroupstaggingapi.ResourceGroupsTaggingAPI
-	flywheel *flywheel.Manager
-	org      string
+	server      *server
+	sp          *sessionParams
+	docdbClient docdb.DocDB
+	rgClient    *resourcegroupstaggingapi.ResourceGroupsTaggingAPI
 }
 
-func newDocDBOrchestrator(client docdb.DocDB, rgclient *resourcegroupstaggingapi.ResourceGroupsTaggingAPI, flywheel *flywheel.Manager, org string) *docDBOrchestrator {
+// sessionParams stores all required parameters to initialize the connection session
+type sessionParams struct {
+	role         string
+	inlinePolicy string
+	policyArns   []string
+}
+
+// newDocDBOrchestrator creates a new session and initializes all clients
+func (s *server) newDocDBOrchestrator(ctx context.Context, sp *sessionParams) (*docDBOrchestrator, error) {
+	log.Debug("initializing docDBOrchestrator")
+
+	sess, err := s.assumeRole(
+		ctx,
+		s.session.ExternalID,
+		sp.role,
+		sp.inlinePolicy,
+		sp.policyArns...,
```

Comment:
We could update `assumeRole` to just take a `sessionParam` here, but I just left it for now